docs/reference/ingest/processors/data-stream-router.asciidoc

```diff
@@ -0,0 +1,40 @@
+[[data-stream-router-processor]]
+=== Data stream router processor
+++++
+<titleabbrev>Data stream router</titleabbrev>
+++++
+
+The `data_stream_router` processor allows to route a document to another data stream.
+It can use both static values or values from the document to determine the target data stream.
+
+The name of a data stream is comprised of three parts and looks like this: `<type>-<dataset>-<namespace>`.
+See the {fleet-guide}/data-streams.html#data-streams-naming-scheme[data stream naming scheme] documentation for more details.
+
+After a `data_stream_router` processor has been executed, all the other processors of the current pipeline are skipped.
+This means that at most one `data_stream_router` processor is ever executed within a pipeline,
```

Comment:
If the data is then delivered to the next data stream, will it be possible to have their another data_stream_router?

Example:

* We detect it is an nginx image, route to `logs-nginx-default`
* On this data stream, there is another data stream router to decide if it is access or error logs. If none match, it is stores in `logs-nginx-default`, otherwise sent to `logs-nginx.access-default` where there is likely and ingest processor to take the message apart.

I remember there were historically some problems around this.

Reply:
Yes, this should be possible. But I didn't explicitly test that scenario yet. I'll make sure to add tests for that if we decide to move forward with this approach.

Reply:
I've added automated tests for this scenario. See `300_data_stream_processor.yml`.

Reply:
Turned out there's an issue with that. 

https://github.com/elastic/elasticsearch/pull/59522
> default pipeline from new index won't be executed

So currently, multi-stage routing is not working. It seems that's due to this concern mentioned in https://github.com/elastic/elasticsearch/issues/57968
> how do we prevent the document from being infinitely redirected between indices

I took a stab at allowing default pipelines of the new index to be executed in [f7c2c51](https://github.com/elastic/elasticsearch/pull/76511/commits/f7c2c510fa0e572b17b637889499c2ada25dd6d3).

Reply:
I've extracted this to a separate PR:
* https://github.com/elastic/elasticsearch/pull/85931